src/pkgTypes.ts

```diff
@@ -0,0 +1,97 @@
+import type { RawPkg, StyleType } from './@types/pkg';
+import type { File } from './jsDelivr';
+import { datadog } from './utils/datadog';
+
+const styleFileExtensions: StyleType[] = ['css', 'less', 'scss'];
+const jsFileExtensions = ['js', 'mjs', 'cjs'];
+
+function createFilePattern(extensions: string[]): RegExp {
+  const extPattern = extensions.join('|');
+  return new RegExp(
+    `^(?:(?!\\/(docs?|documentation|examples?|samples?|demos?|tests?)\\/)(?!\\/[._]).)+\\.(${extPattern})$`
+  );
+}
+
+export function getStyleTypes(
+  pkg: Pick<RawPkg, 'styleTypes'>,
+  filelist: File[]
+): Pick<RawPkg, 'styleTypes'> {
+  const start = Date.now();
+
+  try {
+    const styleTypes = new Set(pkg.styleTypes);
+    const filePattern = createFilePattern(styleFileExtensions);
```

Comment:
I don't you need to build the file pattern at each execution, just a constant is enough no?

Reply:
fixed in https://github.com/algolia/npm-search/commit/8ff01012a86471b8ba3a013efd7c8731f1557ee8